bootstrap/pkg/apis/apps/configconverters/v1beta1.go

```diff
@@ -136,16 +137,13 @@ func (v V1beta1) ToKfConfig(kfdefBytes []byte) (*kfconfig.KfConfig, error) {
 		s := kfconfig.Secret{
 			Name: secret.Name,
 		}
-		if secret.SecretSource == nil {
+		// We don't want to store literalSource explictly, becasue we want the config to be checked into source control and don't want secrets in source control.
+		if secret.SecretSource == nil || secret.SecretSource.LiteralSource != nil {
```

Comment:
add || secret.SecretSource.hashedSource != nil

as well because we don't want to serialize hashed sources either

Reply:
I think hashed source is deprecated?

Reply:
HashedSource is not a field in SecretSource. We don't need to check that.
https://github.com/kubeflow/kubeflow/blob/f943808a2805d81c90935f77b89b8bb85e43baa0/bootstrap/pkg/apis/apps/kfdef/v1beta1/application_types.go#L89

Reply:
Oh great